meson.build

```diff
@@ -521,29 +521,31 @@ features += {'vt.h': cc.has_header_symbol('sys/vt.h', 'VT_GETMODE')}
 
 features += {'consio.h': not features['vt.h'] and cc.has_header_symbol('sys/consio.h', 'VT_GETMODE')}
 
-fragments = join_paths(source_root, 'waftools', 'fragments')
+features += {'osx-thread-name': darwin and cc.has_function('pthread_setname_np', dependencies: pthreads,
```

Comment:
by adding `darwin and` you essentially hardcoded it
the issue between osx <-> glibc is that the function name is identical but it has a different signature, `cc.has_function` can by definition not handle this

Reply:
Yeah, if you don't want to check that the function exists, but rather want to check what its arguments are, you need hand-rolled test files.

Reply:
For our purposes, we really only need to know if it's darwin because macOS is the sole special snowflake here when setting a pthread name (it's not used for anything else). The `has_function` bit is superfluous really since it's not like that should ever fail because we already require `threads` anyways.